### PR TITLE
python3Packages.trytond: 6.2.6 -> 6.4.0

### PR DESCRIPTION
**Pull request by dotlambda**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

cc @udono

### CODE REVIEW

**Comment by gador** [comment]
Could you elaborate a bit on why you had to add a new check dependency to flask-security-too?

**Comment by dotlambda** [comment]
> Could you elaborate a bit on why you had to add a new check dependency to flask-security-too?

It was removed from passlib's `propagatedBuildInputs`. I'm just following what upstream flask-security-too does: https://github.com/Flask-Middleware/flask-security/blob/a3af228a2067db770152289c6d98a564553ff107/requirements/tests.txt#L9

**Comment by gador** [comment]
> > Could you elaborate a bit on why you had to add a new check dependency to flask-security-too?
> 
> It was removed from passlib's `propagatedBuildInputs`. I'm just following what upstream flask-security-too does: https://github.com/Flask-Middleware/flask-security/blob/a3af228a2067db770152289c6d98a564553ff107/requirements/tests.txt#L9

Thanks, that seems sensible.
But why do you include this change in the PR to update trytond? AFAIK it doesn't even depend on Flask-Security-Too. Did the build fail otherwise?

**Comment by dotlambda** [comment]
@gador I guess I should have called the PR "random Python stuff". I just started with trytond and then saw that passlib doesn't specify extras-require even though trytond depends on passlib[bcrypt] and passlib[argon2] so I went ahead and changed that.

**Comment by dotlambda** [comment]
@ofborg eval

**Comment by dotlambda** [comment]
Result of `nixpkgs-review pr 171354` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>16 packages failed to build:</summary>
  <ul>
    <li>dyndnsc</li>
    <li>napalm</li>
    <li>odoo</li>
    <li>privacyidea</li>
    <li>python310Packages.ldaptor</li>
    <li>python310Packages.mailman</li>
    <li>python310Packages.mailman-hyperkitty</li>
    <li>python310Packages.napalm</li>
    <li>python310Packages.napalm-hp-procurve</li>
    <li>python310Packages.privacyidea-ldap-proxy</li>
    <li>python39Packages.json-logging</li>
    <li>python39Packages.ldaptor</li>
    <li>python39Packages.napalm</li>
    <li>python39Packages.napalm-hp-procurve</li>
    <li>python39Packages.privacyidea-ldap-proxy</li>
    <li>zeroc-ice-cpp11</li>
  </ul>
</details>
<details>
  <summary>95 packages built:</summary>
  <ul>
    <li>alerta-server</li>
    <li>buildbot (python39Packages.buildbot)</li>
    <li>buildbot-full (python39Packages.buildbot-full)</li>
    <li>buildbot-ui (python39Packages.buildbot-ui)</li>
    <li>buildbot-worker (python39Packages.buildbot-worker)</li>
    <li>cura</li>
    <li>datadog-agent</li>
    <li>datadog-process-agent</li>
    <li>devpi-client</li>
    <li>devpi-server</li>
    <li>etebase-server (python39Packages.etebase-server)</li>
    <li>etesync-dav</li>
    <li>gnome-keysign</li>
    <li>gns3-gui</li>
    <li>gns3-server</li>
    <li>grab-site</li>
    <li>home-assistant-component-tests.roomba</li>
    <li>home-assistant-component-tests.sentry</li>
    <li>magic-wormhole (python39Packages.magic-wormhole)</li>
    <li>mailman</li>
    <li>mitmproxy (python39Packages.mitmproxy)</li>
    <li>moodle-dl</li>
    <li>moz-phab</li>
    <li>murmur</li>
    <li>netbox</li>
    <li>octoprint</li>
    <li>openai (python39Packages.openai)</li>
    <li>paperless-ngx</li>
    <li>pgadmin4</li>
    <li>python310Packages.amqtt</li>
    <li>python310Packages.autobahn</li>
    <li>python310Packages.buildbot</li>
    <li>python310Packages.buildbot-full</li>
    <li>python310Packages.buildbot-ui</li>
    <li>python310Packages.buildbot-worker</li>
    <li>python310Packages.channels</li>
    <li>python310Packages.channels-redis</li>
    <li>python310Packages.ciscoconfparse</li>
    <li>python310Packages.daphne</li>
    <li>python310Packages.dependency-injector</li>
    <li>python310Packages.etebase-server</li>
    <li>python310Packages.fastapi</li>
    <li>python310Packages.flask-security-too</li>
    <li>python310Packages.labgrid</li>
    <li>python310Packages.magic-wormhole</li>
    <li>python310Packages.magic-wormhole-mailbox-server</li>
    <li>python310Packages.mitmproxy</li>
    <li>python310Packages.ormar</li>
    <li>python310Packages.passlib</li>
    <li>python310Packages.pg8000</li>
    <li>python310Packages.pypiserver</li>
    <li>python310Packages.relatorio</li>
    <li>python310Packages.roombapy</li>
    <li>python310Packages.slowapi</li>
    <li>python310Packages.testing-postgresql</li>
    <li>python310Packages.trytond</li>
    <li>python310Packages.volvooncall</li>
    <li>python39Packages.amqtt</li>
    <li>python39Packages.autobahn</li>
    <li>python39Packages.channels</li>
    <li>python39Packages.channels-redis</li>
    <li>python39Packages.ciscoconfparse</li>
    <li>python39Packages.daphne</li>
    <li>python39Packages.dependency-injector</li>
    <li>python39Packages.django-rq</li>
    <li>python39Packages.fastapi</li>
    <li>python39Packages.flask-security-too</li>
    <li>python39Packages.labgrid</li>
    <li>python39Packages.magic-wormhole-mailbox-server</li>
    <li>python39Packages.mailman</li>
    <li>python39Packages.mailman-hyperkitty</li>
    <li>python39Packages.ormar</li>
    <li>python39Packages.passlib</li>
    <li>python39Packages.pg8000</li>
    <li>python39Packages.pypiserver</li>
    <li>python39Packages.relatorio</li>
    <li>python39Packages.roombapy</li>
    <li>python39Packages.sentry-sdk</li>
    <li>python39Packages.slowapi</li>
    <li>python39Packages.spacy</li>
    <li>python39Packages.spacy-loggers</li>
    <li>python39Packages.spacy-transformers</li>
    <li>python39Packages.testing-postgresql</li>
    <li>python39Packages.textacy</li>
    <li>trytond (python39Packages.trytond)</li>
    <li>python39Packages.volvooncall</li>
    <li>python39Packages.wandb</li>
    <li>radicale</li>
    <li>radicale2</li>
    <li>skytemple</li>
    <li>soco-cli</li>
    <li>tahoe-lafs</li>
    <li>theharvester</li>
    <li>tribler</li>
    <li>zeroc-ice</li>
  </ul>
</details>

File: pkgs/development/python-modules/twisted/default.nix
Comment (SuperSandro2000) [inline pkgs/development/python-modules/twisted/default.nix:51]: I am currently doing major changes in twisted. This needs to be adjusted after that or I cherry-pick those changes into my PR.

Probably doing the later.

File: pkgs/development/python-modules/trytond/default.nix
Comment (SuperSandro2000) [inline pkgs/development/python-modules/trytond/default.nix:21]: Please remove the flag. If in one python env both trytonds exists python gets really confused.

File: pkgs/development/python-modules/relatorio/default.nix
Comment (SuperSandro2000) [inline pkgs/development/python-modules/relatorio/default.nix:31]: Whats up with pycha? Not packaged?

File: pkgs/development/python-modules/passlib/default.nix
Comment (SuperSandro2000) [inline pkgs/development/python-modules/passlib/default.nix:20]: This is now lost which I think was really important.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/twisted/default.nix
Comment (SuperSandro2000) [inline pkgs/development/python-modules/twisted/default.nix:47]: Going to drop that in my PR because we do not have openssl below 1.1.1b https://github.com/twisted/twisted/blob/trunk/setup.cfg#L88

File: pkgs/development/python-modules/twisted/default.nix
Comment (dotlambda) [reply pkgs/development/python-modules/twisted/default.nix:51]: Why not merge this and you rebase?

File: pkgs/development/python-modules/relatorio/default.nix
Comment (dotlambda) [reply pkgs/development/python-modules/relatorio/default.nix:31]: Yup

File: pkgs/development/python-modules/passlib/default.nix
Comment (dotlambda) [reply pkgs/development/python-modules/passlib/default.nix:20]: Those are extra-requires.

File: pkgs/development/python-modules/trytond/default.nix
Comment (dotlambda) [reply pkgs/development/python-modules/trytond/default.nix:21]: I think trytond should be removed from pythonPackages anyway. I won't remove it though cause people ought to know they have to use packageOverrides

File: pkgs/development/python-modules/twisted/default.nix
Comment (SuperSandro2000) [reply pkgs/development/python-modules/twisted/default.nix:51]: Because my PR needs to target staging because it triggers a rebuild and I want to avoid that someone else needs to resolve the more complicated merge conflict.

File: pkgs/development/python-modules/passlib/default.nix
Comment (SuperSandro2000) [reply pkgs/development/python-modules/passlib/default.nix:20]: Yes but extra-requires currently is not easy to use with propagatedNativeBuildInputs. I am not sure how we want to make sure people actually add both.

File: pkgs/development/python-modules/twisted/default.nix
Comment (dotlambda) [reply pkgs/development/python-modules/twisted/default.nix:51]: So you'll have to wait a few hours before merging, that's fine.

File: pkgs/development/python-modules/passlib/default.nix
Comment (dotlambda) [reply pkgs/development/python-modules/passlib/default.nix:20]: Why is propagatedNativeBuildInputs needed?

Comment (SuperSandro2000) [reply pkgs/development/python-modules/passlib/default.nix:20]: For cross-compilation https://github.com/NixOS/nixpkgs/commit/1cb011217d68bb068a9b9a03f6330d5778c86738

Comment (dotlambda) [reply pkgs/development/python-modules/passlib/default.nix:20]: It's not perfect, but not depending on argon2_cffi at all should also fix cross-compilation.

Comment (dotlambda) [reply pkgs/development/python-modules/passlib/default.nix:20]: cc @legendofmiracles

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 171354` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>14 packages failed to build and already failed to build on hydra master:</summary>
  <ul>
<li>napalm: log was empty</li>
<li>odoo: log was empty</li>
<li>privacyidea: log was empty</li>
<li>python310Packages.ldaptor: log was empty</li>
<li>python310Packages.mailman: log was empty</li>
<li>python310Packages.mailman-hyperkitty: log was empty</li>
<li>python310Packages.napalm: log was empty</li>
<li>python310Packages.napalm-hp-procurve: log was empty</li>
<li>python310Packages.privacyidea-ldap-proxy: log was empty</li>
<li>python39Packages.ldaptor: log was empty</li>
<li>python39Packages.napalm: log was empty</li>
<li>python39Packages.napalm-hp-procurve: log was empty</li>
<li>python39Packages.privacyidea-ldap-proxy: log was empty</li>
<li>zeroc-ice-cpp11: log was empty</li>
  </ul>
</details>
<details>
<summary>
97 packages built:
</summary>
<ul>
<li>alerta-server</li>
<li>buildbot (python39Packages.buildbot)</li>
<li>buildbot-full (python39Packages.buildbot-full)</li>
<li>buildbot-ui (python39Packages.buildbot-ui)</li>
<li>buildbot-worker (python39Packages.buildbot-worker)</li>
<li>cura</li>
<li>datadog-agent</li>
<li>datadog-process-agent</li>
<li>devpi-client</li>
<li>devpi-server</li>
<li>dyndnsc</li>
<li>etebase-server (python39Packages.etebase-server)</li>
<li>etesync-dav</li>
<li>gnome-keysign</li>
<li>gns3-gui</li>
<li>gns3-server</li>
<li>grab-site</li>
<li>home-assistant-component-tests.roomba</li>
<li>home-assistant-component-tests.sentry</li>
<li>magic-wormhole (python39Packages.magic-wormhole)</li>
<li>mailman</li>
<li>mitmproxy (python39Packages.mitmproxy)</li>
<li>moodle-dl</li>
<li>moz-phab</li>
<li>murmur</li>
<li>netbox</li>
<li>octoprint</li>
<li>openai (python39Packages.openai)</li>
<li>paperless-ngx</li>
<li>pgadmin4</li>
<li>python310Packages.amqtt</li>
<li>python310Packages.autobahn</li>
<li>python310Packages.buildbot</li>
<li>python310Packages.buildbot-full</li>
<li>python310Packages.buildbot-ui</li>
<li>python310Packages.buildbot-worker</li>
<li>python310Packages.channels</li>
<li>python310Packages.channels-redis</li>
<li>python310Packages.ciscoconfparse</li>
<li>python310Packages.daphne</li>
<li>python310Packages.dependency-injector</li>
<li>python310Packages.etebase-server</li>
<li>python310Packages.fastapi</li>
<li>python310Packages.flask-security-too</li>
<li>python310Packages.labgrid</li>
<li>python310Packages.magic-wormhole</li>
<li>python310Packages.magic-wormhole-mailbox-server</li>
<li>python310Packages.mitmproxy</li>
<li>python310Packages.ormar</li>
<li>python310Packages.passlib</li>
<li>python310Packages.pg8000</li>
<li>python310Packages.pypiserver</li>
<li>python310Packages.relatorio</li>
<li>python310Packages.roombapy</li>
<li>python310Packages.slowapi</li>
<li>python310Packages.testing-postgresql</li>
<li>python310Packages.trytond</li>
<li>python310Packages.volvooncall</li>
<li>python39Packages.amqtt</li>
<li>python39Packages.autobahn</li>
<li>python39Packages.channels</li>
<li>python39Packages.channels-redis</li>
<li>python39Packages.ciscoconfparse</li>
<li>python39Packages.daphne</li>
<li>python39Packages.dependency-injector</li>
<li>python39Packages.django-rq</li>
<li>python39Packages.fastapi</li>
<li>python39Packages.flask-security-too</li>
<li>python39Packages.json-logging</li>
<li>python39Packages.labgrid</li>
<li>python39Packages.magic-wormhole-mailbox-server</li>
<li>python39Packages.mailman</li>
<li>python39Packages.mailman-hyperkitty</li>
<li>python39Packages.ormar</li>
<li>python39Packages.passlib</li>
<li>python39Packages.pg8000</li>
<li>python39Packages.pypiserver</li>
<li>python39Packages.relatorio</li>
<li>python39Packages.roombapy</li>
<li>python39Packages.sentry-sdk</li>
<li>python39Packages.slowapi</li>
<li>python39Packages.spacy</li>
<li>python39Packages.spacy-loggers</li>
<li>python39Packages.spacy-transformers</li>
<li>python39Packages.testing-postgresql</li>
<li>python39Packages.textacy</li>
<li>trytond (python39Packages.trytond)</li>
<li>python39Packages.volvooncall</li>
<li>python39Packages.wandb</li>
<li>radicale (radicale3)</li>
<li>radicale2</li>
<li>skytemple</li>
<li>soco-cli</li>
<li>tahoe-lafs</li>
<li>theharvester</li>
<li>tribler</li>
<li>zeroc-ice</li>
</ul>
</details>

**Pull request merged**
